src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -333,6 +333,21 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     context.AddSource(hintName, csharpDocument.Text);
                 }
             });
+
+            var hostOutputs = csharpDocuments
+                .Collect()
+                .Combine(allTagHelpers)
+                .WithTrackingName("HostOutputs");
+
+#pragma warning disable RSEXPERIMENTAL004 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+            context.RegisterHostOutput(hostOutputs, (context, pair) =>
```

Comment:
Should this be skipped if cohosting flag isn't set, lest it cause minute perf differences for everyone else?

Reply:
Yeah, its probably not going to be noticeable, but given that we have the suppression flag readily available I'll gate it behind it just in case.